src/core.js

```diff
@@ -227,6 +227,8 @@ const alignWithDOM = function(nodeName, key) {
     const keyNode = keyMap[key];
     if (matches(nodeName, key, keyNode)) {
       node = keyNode;
+    } else if (keyNode && keyNode !== currentNode) {
+      removeChild(currentParent, keyNode, keyMap);
```

Comment:
Nope, that's just the normal DOM API. We can't mark deleted there, because the user may later use the key to grab the element (even though it's no longer in the DOM tree).